.travis.yml

```diff
@@ -3,28 +3,25 @@ services:
 
 before_install:
   - echo "$DOCKER_HUB_PASSWORD" | docker login --username $DOCKER_HUB_USERNAME --password-stdin
-  - curl https://cli-assets.heroku.com/install.sh | sh  # install heroku
-  - docker login --username=$HEROKU_LOGIN --password=$HEROKU_API_KEY registry.heroku.com 
 
 before_script:
   - docker build --target test --tag test .
   - docker build --tag $DOCKER_HUB_USERNAME/todo-app:latest --target production .
-  - docker tag $DOCKER_HUB_USERNAME/todo-app:latest registry.heroku.com/$HEROKU_APP/web
 
 script:
   - docker run test tests
   - docker run test tests_integration
   - >
     docker run
-    -e MONGO_DB_USERNAME
-    -e MONGO_DB_PASSWORD
+    -e MONGO_DB_PRIMARY_CONNECTION_STRING
```

Comment:
The error thrown by travis suggests that this environment variable is being passed through empty - in that case pymongo tries to connect to a locally running mongo instance and the error message mentions `localhost:27017`
To track this down, I'd start by checking the env variable was encrypted correctly - adding the variable via travis' gui & removing it from your config for example may make that quicker/easier to verify

Reply:
@JackMead - Bingo! Thank you, this has been driving me nuts. The build is now succeeding and the E2E test completed successfully on Travis.